Client/Frontend/ContentBlocker/ContentBlockerHelper.swift

```diff
@@ -106,42 +106,56 @@ class ContentBlockerHelper {
         return EnabledState(rawValue: pref) ?? .onInPrivateBrowsing
     }
 
-    fileprivate func addActiveRulesToTab() {
-        guard let ruleStore = ruleStore else { return }
+    func addActiveRulesToTab() -> Deferred<[String]> {
+        let result = Deferred<[String]>()
+        guard let ruleStore = ruleStore else { return result }
         let rules = blocklistBasic + (blockingStrengthPref == .strict ? blocklistStrict : [])
         let enabledMode = enabledStatePref
         removeAllFromTab()
 
         func addRules() {
```

Comment:
I think you can just move this outside of this method. You can then pass in the rules array as a parameter.

Reply:
This would look odd I think, it is an internal utility function (an anonymous func almost) as it has the job of filling the Deferred for addActiveRulesToTab() –the 'oddness' would be the added parameter of passing in the deferred to fill.